### PR TITLE
Add loss function capabilities

### PR DESCRIPTION
**Pull request by dmpelt**
Changes proposed in this pull request:

* Add ability to choose loss function
* Add ability to use custom loss function
* Note: you now have to explicitly choose a loss function in training scripts

Fixes issue #17 (and also negative gradients of #9 )

Checklist:

* [x] Update CHANGELOG.md

### CODE REVIEW

**Comment by keeeto** [comment]
This looks great. Just to check - when we want to define a new loss it should have the methods `loss` and `deriv` in the class?

**Comment by dmpelt** [comment]
Yes, indeed! They both return images, with the loss value of each pixel and the derivative of the loss for each pixel. As an example, this is the L1 loss given in [the example script](https://github.com/dmpelt/msdnet/blob/fbb300c5b2cb97478975b9cd49fd9b14250eb38a/examples/train_regr_customloss.py):

```python
class L1Loss(msdnet.loss.Loss):
    '''Computes L1-norm loss function.'''
    def loss(self, im, tar):
        '''Computes loss function for each pixel.'''
        err = np.zeros_like(tar)
        err[:] = np.abs(tar - im)
        return err

    def deriv(self, im, tar):
        '''Computes derivative of loss function.'''
        err = np.zeros_like(tar)
        err[im<tar] = -1
        err[im>tar] = 1
        return err
```

If there is something you want added/changed/removed in the PR, please let me know! If not, are you OK with merging this PR as it is?

**Comment by keeeto** [comment]
Sorry I didn't reply sooner @dmpelt - this looks great. I am looking forward to trying it out myself.
